### DISCUSSION TITLE
Any plans for virtualisation on iPadOS 16?

### DISCUSSION THREAD

**Discussion by ToneXum** [utmapp/UTM#5748] [UTM Help (iOS)] [answered]
Ive seen a bunch of people getting some VM to work with virtualisation on iPad, but they all run iPadOS 15 or lower because TrollStore supports it.

Ive done some digging and found out that a framework called hypervisor is required for proper virtualisation and that Apple needs to release it for iPadOS 16.

Now, do we know anything about hypervisor.framework coming to iPadOS 16?
(Correct me if Im wrong)

**Comment by Edgecrusherr160** [comment, marked as answer]
Still no hypervisor support in iPadOS / iOS 17. Hypervisor is indeed physically possible on any of the M series chips included in the iPad Airs and iPad Pros, but locked away the iPadOS. Last I checked, the only way you can get access to it is by full jailbreaking, which currently is a struggle past iPadOS 15.

Unfortunately, Apple is very anti-virtualization/emulation on their mobile devices. It's very frustrating, and it's seriously holding the iPad back. My guess is that they're afraid of the possible security risk, but also they don't want to see news headlines of people running Windows and macOS on their iPads. Unfortunately, I think this is mostly a lack of vision on Apple's part, and partly a symptom of the systematic problem of rushing major system updates out every year. I wish they would just take some time and figure out what they can really do with the iPad. My M2 iPad Pro is literally the second most powerful computing device I currently own, yet my daughter's 2012 MacBook Air is far more capable in so many areas, due to not having so many constraints upon the OS. I get that they want the iPad to first and foremost be for basic users, but it doesn't take a lot of imagination to work out ways to give advanced users you the ability to unlock advanced features, and the fact that including an M series chip in the iPad Pro is sort of contradictory to their goals (ignoring for a second that they're real "goals" are to spout out marketing hype at people to spend more money).

The iPad is great, but it needs more fluid multitasking, the ability to access more advance features, such as the hypervisor, and ideally, access to apps outside the App Store (maybe next year, thanks to the European union, we can only hope).

If I could just run UTM on my iPad to get access to macOS and Windows, I would have so much portable computing freedom.

**Reply by osy** [reply]
Apple removed kernel support in iOS https://twitter.com/utmapp/status/1708907045314035986

**Reply by ToneXum** [reply]
Lucky me on iPadOS 16.2

**Reply by mfbolus** [reply]
I wish I had run across this sooner. I wouldn't have upgraded. Has the state of this issue changed at all in the interim? i.e. any loosening or workarounds found? I would assume not. 

@osy  et al, thanks for your work on the project

**Comment by Thekidiacs** [comment]
I have an iPad Pro M1 Chip on iOS 14.8.1 will the Hypervisor work? I’m asking because it’s an older iOS I know it works on 16.3.1 and below and iOS 15 but what about 14?

**Reply by ToneXum** [reply]
Going by the documentation, UTM should work perfectly fine on your version if you have a jailbreak. Even the hypervisor.
